### ISSUE TITLE
Padding-free warning fires for revision-pinned hub kernels (`repo_id@revision`)

### ISSUE THREAD

**Issue by albertvillanova** [huggingface/trl#6984]
### Describe the bug

`SFTTrainer` decides whether the attention implementation supports padding-free training with an exact-match membership test against a hardcoded set of bare repo ids:

```python
# trl/trainer/sft_trainer.py
use_flash_attention = model.config._attn_implementation in FLASH_ATTENTION_VARIANTS
```

Transformers, however, accepts a revision suffix on hub kernels, `repo_id@revision` (and a kernel name, `repo_id:kernel_name`), and keeps the full string in `config._attn_implementation`. So `kernels-community/flash-attn2@v2` misses the set and the trainer warns:

> Padding-free training is enabled, but the attention implementation is not set to a supported Flash Attention variant. [...]

The same applies to the packing check a few lines below.

This matters right now because pinning a kernel revision is the current workaround for https://github.com/huggingface/kernels-community/issues/1085, where the v3 stable-ABI builds (the default resolution since transformers 5.16.1) crash in the backward pass for GQA models. The revision syntax is what the `kernels` maintainer recommends (https://github.com/huggingface/kernels-community/issues/1082#issuecomment), and TRL itself adopts it for the invariant suite in #6979. So TRL currently warns users against its own documented workaround.

### Scope

Warning only. Both uses are `logger.warning`, and the collator's padding-free behavior is driven by `self.padding_free`, so no training behavior changes. It is misleading output, not a silent correctness problem.

### Reproduction

```python
from trl import SFTConfig, SFTTrainer

SFTTrainer(
    model="Qwen/Qwen2.5-0.5B-Instruct",
    args=SFTConfig(
        padding_free=True,
        max_length=None,
        model_init_kwargs={"attn_implementation": "kernels-community/flash-attn2@v2", "dtype": "bfloat16"},
    ),
    train_dataset=...,
)
```

### Expected behavior

A revision-pinned or kernel-name-suffixed hub kernel is recognized as the same implementation as its bare repo id, and no warning is emitted.

### Fix

Compare against the repo id alone, by stripping the `@revision` and `:kernel_name` suffixes before the lookup.

### Notes

- `FLASH_ATTENTION_VARIANTS` is also defined in `trl/trainer/dpo_trainer.py` but is unused there, so DPO has nothing to fix. Worth removing separately.
- Independently of this, the underlying upstream breakage (fine-tuning any GQA model through the flash-attn2 hub kernel, including the silent `flash_attention_2` fallback when the `flash-attn` package is not installed) deserves a documentation note. Tracked separately.
